test/test-thread-priority.c

```diff
@@ -0,0 +1,117 @@
+/* Copyright libuv contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h> /* memset */
+
+#ifdef __POSIX__
+#include <pthread.h>
+#include <errno.h>
+#endif
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+uv_sem_t sem;
+
+static void simple_task(void *args) {
+    uv_sem_wait(&sem);
+    printf("in simple_task\n");
+}
+
+TEST_IMPL(thread_priority) {
+  int priority;
+  int r;
+#ifndef _WIN32
+  int min;
+  int max;
+  int policy;
+  struct sched_param param;
+#endif
+  uv_thread_t task_id;
+
+  /* Verify that passing a NULL pointer returns UV_EINVAL. */
+  ASSERT_EQ(UV_EINVAL, uv_thread_getpriority(0, NULL));
+
+  r = uv_sem_init(&sem, 1);
+  ASSERT(r == 0);
```

Comment:
Here and elsewhere:
```suggestion
  ASSERT_OK(r);
```
Having said that, it's a little nicer to write it like this:
```c
ASSERT_OK(uv_sem_init(&sem, 1));
```
Because then you see the operation that failed instead of just `r`.

Reply:
Updated